src/sonic-py-common/sonic_py_common/interface.py

```diff
@@ -88,5 +105,47 @@ def get_port_table_name(interface_name):
         return "VLAN_INTERFACE"
     elif interface_name.startswith(loopback_prefix()):
         return "LOOPBACK_INTERFACE"
+    elif VLAN_SUB_INTERFACE_SEPARATOR in interface_name:
+        if interface_name.startswith(physical_subinterface_prefix()) or interface_name.startswith(portchannel_subinterface_prefix()):
+            return "VLAN_SUB_INTERFACE"
     else:
         return ""
+
+def subintf_get_longname(intf):
+    if intf is None:
+       return None
+    sub_intf_sep_idx = intf.find(VLAN_SUB_INTERFACE_SEPARATOR)
+    if sub_intf_sep_idx == -1:
+        return str(intf)
+    parent_intf = intf[:sub_intf_sep_idx]
+    sub_intf_idx = intf[(sub_intf_sep_idx+1):]
+    if intf.startswith("Eth"):
+        intf_index=intf[len("Eth"):sub_intf_sep_idx]
+        return "Ethernet"+intf_index+VLAN_SUB_INTERFACE_SEPARATOR+sub_intf_idx
+    elif intf.startswith("Po"):
+        intf_index=intf[len("Po"):sub_intf_sep_idx]
+        return "PortChannel"+intf_index+VLAN_SUB_INTERFACE_SEPARATOR+sub_intf_idx
+    else:
+        return str(intf)
+
+def intf_get_longname(intf):
```

Comment:
please follow a similar naming convention in the file.. suggest rename to `get_subintf_name`

Reply:
Since this is being used to get longname for both subinterface and parent of subinterface, I have updated to use get_intf_longname as part of next commit.